bulk_deploy/tsuru_api_client.rb

```diff
@@ -0,0 +1,438 @@
+require "net/http"
+require "uri"
+require "cgi"
+require "json"
+require "yajl"
+require "openssl"
+
+class TsuruAPIClient
+
+  # Default headers added to each request
+  HEADERS = {
+    "Content-Type"    => "application/json",
+    "Accept-Encoding" => "gzip",
+  }
+
+  VERB_MAP = {
+    :get    => Net::HTTP::Get,
+    :post   => Net::HTTP::Post,
+    :put    => Net::HTTP::Put,
+    :delete => Net::HTTP::Delete
+  }
+
+  def initialize(logger:, environment:, host:, protocol: "https://")
+    # Parse the Tsuru API domain and create a HTTP client
+    uri = URI.parse(protocol + environment + "-api." + host)
+    @http = Net::HTTP.new(uri.host, uri.port)
+
+    # Set SSL flag to true if protocol is HTTPS
+    if protocol == "https://"
+      @http.use_ssl = true
+      @http.verify_mode = OpenSSL::SSL::VERIFY_NONE
```

Comment:
`@http` is an instance variable for `TsuruAPIClient`, so I'd only expect it to connect to the API and be presented a valid certificate.

If we need to make requests to applications at a later date then I'd expect that to go into a separate class/module. In which case we should disable verification there, only, with a reference to [#96758396](https://www.pivotaltracker.com/story/show/96758396) so that we can fix it later.